### PR TITLE
Updated server with v1beta1 API

### PR DESCRIPTION
**Pull request by aysylu**
This is the server implementation for the `0.1.0` release.

### CODE REVIEW

File: go/config/config.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+
+	fernet "github.com/fernet/fernet-go"
+	"gopkg.in/yaml.v2"
+)
+
+// File is the grafeas config file.
```
Comment (vtsao) [inline go/config/config.go]: File should be lowercase, comment should always match the method or type name.

File: go/config/config.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	User     string `yaml:"user"`
+	Password string `yaml:"password"`
+	// Valid sslmodes: disable, allow, prefer, require, verify-ca, verify-full.
+	// See https://www.postgresql.org/docs/current/static/libpq-connect.html for details
+	SSLMode       string `yaml:"sslmode"`
+	PaginationKey string `yaml:"paginationkey"`
+}
+
+// Config is the global configuration for an instance of Grafeas.
+type config struct {
```
Comment (vtsao) [inline go/config/config.go]: // config...

File: go/config/config.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	API            *Config              `yaml:"api"`
+	StorageType    string               `yaml:"storage_type"` // Supported storage types are "memstore", "postgres" and "embedded"
+	PgSQLConfig    *PgSQLConfig         `yaml:"postgres"`
+	EmbeddedConfig *EmbeddedStoreConfig `yaml:"embedded"` // EmbeddedConfig is the embedded store config
+}
+
+// DefaultConfig is a configuration that can be used as a fallback value.
+func defaultConfig() *config {
```
Comment (vtsao) [inline go/config/config.go]: // defaultConfig...

File: go/config/config.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+			CAFile:   "",
+		},
+		StorageType: "memstore",
+		PgSQLConfig: &PgSQLConfig{},
+	}
+}
+
+// Creates a config from a YAML-file. If fileName is an empty
+// string a default config will be returned.
+func LoadConfig(fileName string) (*config, error) {
```
Comment (vtsao) [inline go/config/config.go]: Comments on exported methods should always start with the method name. E.g.,
// LoadConfig creates a ...

File: go/config/config.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	if fileName == "" {
+		return defaultConfig(), nil
+	}
+	data, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	var configFile file
+	err = yaml.Unmarshal(data, &configFile)
+	if err != nil {
```
Comment (vtsao) [inline go/config/config.go]: Inline with line above.

File: go/v1beta1/api/noop_auth.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+// NoOpAuth does nothing.
+type NoOpAuth struct{}
+
+func (a *NoOpAuth) CheckAccessAndProject(ctx context.Context, projectID string, entityID string, p iam.Permission) error {
+	return nil
+}
+
+func (a *NoOpAuth) EndUserID(ctx context.Context) (string, error) {
+	return "22", nil
```
Comment (vtsao) [inline go/v1beta1/api/noop_auth.go]: Maybe return "0" instead?

File: go/v1beta1/main/init.go [hunk 7]
```diff
@@ -0,0 +1,223 @@
+// Copyright 2017 The Grafeas Authors. All rights reserved.
+//
```
Comment (vtsao) [inline go/v1beta1/main/init.go]: Can we just combine init.go with main.go, it makes it easier to see what's happening in main.

File: go/config/config.go (hunk 1 above)
Comment (aysylu) [reply go/config/config.go]: good point, cleaned up this file

File: go/config/config.go (hunk 4 above)
Comment (aysylu) [reply go/config/config.go]: good point, did a cleanup

File: go/v1beta1/main/init.go (hunk 7 above)
Comment (aysylu) [reply go/v1beta1/main/init.go]: good idea

**Comment by aysylu** [comment]
@vtsao all the outstanding comments have been addressed, so I'll merge this, but happy to discuss any other feedback and open another PR to address it.

**Pull request merged**